src/main/java/apoc/util/Util.java

```diff
@@ -686,4 +691,99 @@ public static boolean classExists(String className) {
             return null;
         }
     }
+
+    public static Stream<MapResult> runWithRetry( GraphDatabaseService db, Log log, String statement, long retries, List<String> additionalRetryCodes,
+            Map<String,Object> params )
+    {
+        String transactionFailureMessage = "";
+        Long timesTried = 0L;
+        do
+        {
+
+            FutureTask<Result> resultFuture = new FutureTask<>( () -> {
+                try ( Transaction tx = db.beginTx() )
+                {
+                    Result result = ((Function<Map<String,Object>,Result>) ( Map<String,Object> p ) -> db.execute( withParamMapping( statement, p.keySet() ),
```

Comment:
what is wrong with just calling it?

Reply:
I was thinking that we only want the result once the query has finished. By putting it in a FutureTask I was thinking we would only return the result once the entire query was done. I will remove the FutureTask and just run it.

Reply:
I was changing it back and I realized why I had it as a FutureTask rather than just calling it. 

I did this because if the Tx fails, it marks every transaction created in the thread for `rollback`. So even if the retry is successful the transaction is always rolled back. 

This is why I put it in a FutureTask object and then ran it in a separate thread each time.